src/OpenLoco/Map/TileLoop.hpp

```diff
@@ -97,13 +95,16 @@ namespace OpenLoco::Map
         };
 
     public:
-        TilePosRangeView(const TilePos2& begin, const TilePos2& end)
-            : _begin(begin)
-            , _end(end)
+        TilePosRangeView(const TilePos2& bottomLeft, const TilePos2& topRight)
+            : _bottomLeft(bottomLeft)
+            , _topRight(topRight)
         {
         }
 
-        Iterator begin() { return Iterator(_begin, _end); }
-        Iterator end() { return Iterator(_end, _end); }
+        Iterator begin() { return Iterator(_bottomLeft, _topRight); }
+        Iterator end()
+        {
+            return Iterator(TilePos2(_bottomLeft.x, _topRight.y + 1), _topRight); // End iterator must be 1 step past the end so that loop is inclusive
```

Comment:
```suggestion
            // End iterator must be 1 step past the end so that loop is inclusive
            return Iterator(TilePos2(_bottomLeft.x, _topRight.y + 1), _topRight);
```